File: TELERIK ALGORITHMS/TELERIK DSA HW5/LargeCollectionOfProducts/Item.cs
namespace LargeCollectionOfProducts
{
    using System;

    class Item : IComparable<Item>
    {
        private string name;
        private int price;

        public string Name
        {
            get { return this.name; }
            private set { }
        }

        public int Price
        {
            get { return this.price; }
            private set { }
        }

        public Item(string name, int price)
        {
            this.name = name;
            this.price = price;
        }

        public int CompareTo(Item oderItem)
        {
            return this.price.CompareTo(oderItem.price);
        }
    }
}

File: TELERIK-JS-OOP/TELERIK-OOP-EXAPMPREPARATION/TELERIK-OOP-EXAPMPREPARATION/item.js
define(function () {
    var Item;
    Item = (function () {
     
        function Item(type, name, price) {
            if (type == 'smart-phone' || type == 'accessory' || type == 'notebook' || type == 'pc' || type == 'tablet') {
                this.type = type;
            }
            else {
                throw {
                    massage:'Unexisting type of item'
                }
            }
            if (name.length > 40 || name.length < 6) {
                throw {
                    massage : 'Item name must be between 6 and 40 symbols'
                }
            }
            else {
                this.name = name;
            }
           
            this.price = price;
        }
        return Item;
    }());
    return Item;
})